dds/DCPS/XTypes/DynamicDataImpl.cpp

```diff
@@ -302,7 +302,7 @@ DynamicDataImpl::SingleValue::SingleValue(ACE_OutputCDR::from_boolean value)
 {}
 
 DynamicDataImpl::SingleValue::SingleValue(const char* str)
-  : kind_(TK_STRING8), str_(ACE_OS::strdup(str))
+  : kind_(TK_STRING8), active_(&str_), str_(ACE_OS::strdup(str))
```

Comment:
```suggestion
  : kind_(TK_STRING8), active_(0), str_(ACE_OS::strdup(str))
```
`active_` pointer isn't used for this member.